apps/src/appsApi.js

```diff
@@ -0,0 +1,213 @@
+ */
+
+/* jshint
+ funcscope: true,
+ newcap: true,
+ nonew: true,
+ shadow: false,
+ unused: true,
+
+ maxlen: 90,
+ maxparams: 3,
+ maxstatements: 200
+ */
+/* global $ */
+/* global JSON */
+'use strict';
+
+/** Namespace for app storage. */
+var appsApi = module.exports;
+
+/**
+ * Accessor object for an app's remote storage space.
+ * @constructor
+ */
+appsApi.AppStorage = function () {
+  this.apiBaseUrl_ = '/v3/apps';
+};
+
+/**
+ * @param {!function} callback
+ */
+appsApi.AppStorage.prototype.readAll = function (callback) {
+  $.ajax({
+    url: this.apiBaseUrl_,
+    type: 'get',
+    dataType: 'json'
+  }).done(function (data /*, text*/) {
+    callback(data);
+  }).fail(function (/*request, status, error*/) {
+    callback(null);
+  });
+};
+
+/**
+ * @param {!string} id
+ * @param {!function} callback
+ */
+appsApi.AppStorage.prototype.read = function (id, callback) {
+  $.ajax({
+    url: this.apiBaseUrl_ + '/' + id,
+    type: 'get',
+    dataType: 'json'
+  }).done(function (data /*, text*/) {
+    callback(data);
+  }).fail(function (/*request, status, error*/) {
+    callback(undefined);
+  });
+};
+
+/**
+ * @param {!Object} value
+ * @param {!function} callback
+ */
+appsApi.AppStorage.prototype.create = function (value, callback) {
+  $.ajax({
+    url: this.apiBaseUrl_,
+    type: 'post',
+    contentType: 'application/json; charset=utf-8',
+    data: JSON.stringify(value)
+  }).done(function (data /*, text*/) {
+    callback(data);
+  }).fail(function (/*request, status, error*/) {
+    callback(undefined);
+  });
+};
+
+/**
+ * @param {!string} id
+ * @param {!Object} value
+ * @param {!function} callback
+ */
+appsApi.AppStorage.prototype.update = function (id, value, callback) {
+  $.ajax({
+    url: this.apiBaseUrl_ + '/' + id,
+    type: 'post',
+    contentType: 'application/json; charset=utf-8',
+    data: JSON.stringify(value)
+  }).done(function (/*data, text*/) {
+    callback(true);
+  }).fail(function (/*request, status, error*/) {
+    callback(false);
+  });
+};
+
+/**
+ * @param {!string} id
+ * @param {!function} callback
+ */
+appsApi.AppStorage.prototype.delete = function (id, callback) {
+  $.ajax({
+    url: this.apiBaseUrl_ + '/' + id + '/delete',
+    type: 'post',
+    dataType: 'json'
+  }).done(function (/*data, text*/) {
+    callback(true);
+  }).fail(function (/*request, status, error*/) {
+    callback(false);
+  });
+};
+
+/**
+ * API for interacting with app property bags on the server.
+ * This property bag is shared between all users of the app.
+ *
+ * @param app_publickey
+ * @constructor
+ * @augments appsApi.AppStorage
+ */
+appsApi.SharedPropertyBag = function (app_publickey) {
+  appsApi.AppStorage.call(this);
+
+  /** For the most part, property bags just use a different base URL. */
+  this.apiBaseUrl_ = '/v3/apps/' + app_publickey + '/shared-properties';
+};
+appsApi.SharedPropertyBag.prototype = Object.create(appsApi.AppStorage.prototype);
```

Comment:
This inheritance relationship seems overly ambitious because a SharedPropertyBag should not have create, read, update and delete methods. I believe this is the "refused bequest" code smell: http://en.wikipedia.org/wiki/Code_smell

I would propose that the SharedPropertyBag "has an" AppStorage (rather than "is an").